### PR TITLE
feat: add `--client-need-client-entry` and `--client-need-hot-entry`

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [x] This is a **feature**
- [ ] This is a **code refactor**
- [x] This is a **test update**
- [x] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Yes.

### Motivation / Use-Case

Addded -

- `--client-need-client-entry`
-  `--no-client-need-client-entry`
- `--client-need-hot-entry`
- `--no-client-need-hot-entry`

### Breaking Changes
None

### Additional Info
No

### CODE REVIEW

File: bin/cli-flags.js [hunk 1]
```diff
@@ -123,6 +123,41 @@ module.exports = {
         delete opts.clientOverlay;
       },
     },
+    {
+      name: 'client-need-client-entry',
```
Comment (snitin315) [inline bin/cli-flags.js:127]: Maybe we need a better name, double `client` can be confusing.

Comment (alexander-akait) [reply bin/cli-flags.js:127]: Yep, good catch I will think about it

Comment (snitin315) [reply bin/cli-flags.js:127]: Maybe: 

```js
client: {
   appendDefaultEntries: true,
   appendHotEntries: true
}
```

Comment (alexander-akait) [reply bin/cli-flags.js:127]: hm, give me time to think about these options, maybe we need `{ client: (entry) => { return clientOption; } }`, so you will setup hot/client entry for multiple entrypoints

**Review by evenstensberg — APPROVED** [review]

**Comment by alexander-akait** [comment]
@snitin315 I found solution:
- let's rename `needHotEntry` -> `hotEntry: boolean` (`--client-hot-entry`), default value `const hotEntry = options.client.hotEntry || options.hot`
- let's remove `needClientEntry` in favor `client: boolean | ClientOptions | function(entryName): ClientOptions`, so you can disable `hotEntry` or whole client by entries using `function`

Maybe in future we will allow to write to only specific client (we already have this information internally, nobody ask this, so let's postpone)

**Comment by snitin315** [comment]
Nice solution. I will update 👍

**Pull request closed without merging**